cli/pkg/kctrl/local/sources/git.go

```diff
@@ -11,62 +11,61 @@ import (
 	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
 )
 
-type GitStep struct {
-	ui           cmdcore.AuthoringUI
-	vendirConfig vendirconf.Config
+type SourceConfiguration interface {
+	Configure() error
 }
 
-func NewGitStep(ui cmdcore.AuthoringUI, vendirConfig vendirconf.Config) *GitStep {
-	return &GitStep{
-		ui:           ui,
-		vendirConfig: vendirConfig,
-	}
+type GitConfiguration struct {
+	ui           cmdcore.AuthoringUI
+	vendirConfig *VendirConfig
 }
 
-func (g *GitStep) PreInteract() error { return nil }
+var _ SourceConfiguration = &GitConfiguration{}
 
-func (g *GitStep) Interact() error {
-	contents := g.vendirConfig.Directories[0].Contents
+func NewGitConfiguration(ui cmdcore.AuthoringUI, vendirConfig *VendirConfig) *GitConfiguration {
+	return &GitConfiguration{ui: ui, vendirConfig: vendirConfig}
+}
+
+func (g *GitConfiguration) Configure() error {
+	contents := g.vendirConfig.Contents()
 	if contents == nil {
-		err := g.initializeContentWithGit()
+		err := g.initializeContentWithGit(contents)
 		if err != nil {
 			return err
 		}
 	} else if contents[0].Git == nil {
-		err := g.initializeGit()
+		err := g.initializeGit(contents)
 		if err != nil {
 			return err
 		}
 	}
 
-	err := g.configureGitURL()
+	err := g.configureGitURL(contents)
 	if err != nil {
 		return err
 	}
-	err = g.configureGitRef()
+	err = g.configureGitRef(contents)
 	if err != nil {
 		return err
 	}
-	return g.getIncludedPaths()
+	return g.getIncludedPaths(contents)
 }
 
-func (g *GitStep) PostInteract() error { return nil }
-
-func (g *GitStep) initializeContentWithGit() error {
+func (g *GitConfiguration) initializeContentWithGit(contents []vendirconf.DirectoryContents) error {
 	//TODO Rohit need to check this how it should be done. It is giving path as empty.
-	g.vendirConfig.Directories[0].Contents = append(g.vendirConfig.Directories[0].Contents, vendirconf.DirectoryContents{})
```

Comment:
In the configure we have contents as `g.vendirConfig.Contents()` but here it is `g.vendirConfig.Directories[0].Contents`, is this change intentional?